src/components/SuperLikeContentCard.vue

```diff
@@ -0,0 +1,247 @@
+    },
+    superLikeShortId: {
+      type: String,
+      default: undefined,
+    },
+    timestamp: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data() {
+    return {
+      isLoading: true,
+
+      author: { user: this.authorId },
+      internalUrl: this.referrer,
+      internalTitle: this.title,
+      internalDescription: this.description,
+      internalCoverSrc: this.coverSrc,
+      internalLikeCount: this.likeCount,
+    };
+  },
+  computed: {
+    ...mapGetters([
+      'getUserId',
+      'getUserInfoById',
+      'getArticleInfoByReferrer',
+      'getIsInBookmark',
+      'getIsFollowedAuthor',
+    ]),
+
+    shouldFetchArticle() {
+      return !this.internalTitle;
+    },
+    href() {
+      if (this.superLikeShortId) {
+        return getSuperLikeRedirectLink(this.superLikeShortId);
+      }
+      if (this.superLikeId) return getSuperLikeRedirectLink(this.superLikeId);
+      return this.internalUrl;
+    },
+  },
+
+  methods: {
+    ...mapActions([
+      'fetchUserInfo',
+      'fetchArticleInfo',
+      'addBookmark',
+      'removeBookmark',
+      'followAuthor',
+      'unfollowAuthor',
+    ]),
+
+    formatDate(timestamp) {
+      return dateFormat(new Date(timestamp), 'DD/MM/YYYY');
+    },
+
+    async fetchContent() {
+      try {
+        this.isLoading = true;
+        const promises = [];
+        if (
+          this.shouldFetchArticle &&
+          !this.getArticleInfoByReferrer(this.referrer)
+        ) {
+          promises.push(
+            this.fetchArticleInfo(this.referrer)
+              .then(() => this.updateArticleInfo())
+              .catch(() => ({}))
+          );
+        } else {
+          this.updateArticleInfo();
+        }
+        if (!this.author.user) await Promise.all(promises);
+        if (this.author.user && !this.getUserInfoById(this.author.user)) {
+          promises.push(
+            this.fetchUserInfo(this.author.user)
+              .then(() => this.updateAuthorInfo())
+              .catch(() => ({}))
+          );
+        } else {
+          this.updateAuthorInfo();
+        }
+        await Promise.all(promises);
+      } finally {
+        this.$emit('fetched');
+        this.isLoading = false;
+      }
+    },
+    updateArticleInfo() {
+      if (this.getArticleInfoByReferrer(this.referrer)) {
+        const {
+          url,
+          image,
+          title,
+          description,
+          like,
+          user,
+        } = this.getArticleInfoByReferrer(this.referrer);
+        this.internalUrl = url;
+        this.internalTitle = title;
+        this.internalDescription = description;
+        this.internalCoverSrc = image;
+        this.internalLikeCount = like;
+        // If authorId is not given, we can retrieve from article info
+        if (!this.author.user) {
+          this.author = { user };
+        }
+      }
+    },
+    updateAuthorInfo() {
+      this.author = this.getUserInfoById(this.author.user) || this.author;
```

Comment:
Would having `authorId` as string data, computed prop based on `authorId` that returns `getUserInfoById `, and a watcher on `authorId` to fetch user data if not exists be a easier to understand structure than this union?

Reply:
I don't know, this component is duplicated from ContentCardWrapper, and you wrote this line:
https://github.com/likecoin/liker-land/blame/258487ea565402c8097f88b6a419b7a737c2355d/src/components/ContentCardWrapper.vue#L177

Reply:
Maybe we(i) can refactor this part later